crates/core/src/client/helpers.rs

```diff
@@ -105,18 +109,16 @@ pub fn decode_eth_call_return<T: std::error::Error>(
     // fail is Kakarot API changes)
     // Declare Vec of Result
     let mut segmented_result: Vec<FeltOrFeltArray> = Vec::new();
-    let mut tmp_array_len: FieldElement = *call_result.first().ok_or_else(|| {
-        EthApiError::OtherError(anyhow::anyhow!("Cannot parse and decode return arguments of Kakarot call",))
-    })?;
+    let mut tmp_array_len: FieldElement = *call_result
```

Comment:
I think that these functions are not accurate anymore since `eth_call` and `eth_send_transaction` return only `return_data_len: felt, return_data: felt*`

Reply:
only the comments were wrong indeed, because it does read the first return array, which is the return data. So plz update the string, there is no stack here